File: Problem-Solving/sparseArr.c
#include <stdio.h>
#include <string.h>
int main() {

  int strings_count, queries_count, count = 0, i, j, s;
  char strings[10000][20], queries[10000][20];
  scanf("%d", &strings_count);
  for (j = 0; j < strings_count; j++) {
    scanf("%s", strings[j]);
  }
  scanf("%d", &queries_count);
  for (i = 0; i < queries_count; i++) {
    scanf("%s", queries[i]);
  }
  for (i = 0; i < queries_count; i++) {
    count = 0;
    for (j = 0; j < strings_count; j++) {
      s = strcmp(queries[i], strings[j]);
      if (s == 0) {
        count = count + 1;
      }
    }
    printf("%d\n",count);
  }
return 0;
}

File: Problem-Solving/Python/sparse-arr.py
#!/bin/python3
import os

def matchingStrings(strings, queries):
    res = []
    for query in queries:
        if query in strings:
            count = strings.count(query)
            res.append(count)
        else:
            res.append(0)
    return res

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    strings_count = int(input())

    strings = []

    for _ in range(strings_count):
        strings_item = input()
        strings.append(strings_item)

    queries_count = int(input())

    queries = []

    for _ in range(queries_count):
        queries_item = input()
        queries.append(queries_item)

    res = matchingStrings(strings, queries)

    fptr.write('\n'.join(map(str, res)))
    fptr.write('\n')

    fptr.close()